http/action/src/main/java/io/knotx/databridge/http/action/HttpActionOptions.java

```diff
@@ -94,6 +95,44 @@ public HttpActionOptions setRequestTimeoutMs(long requestTimeoutMs) {
     return this;
   }
 
+  public String getResponseType() {
+    return responseType;
+  }
+
+  /**
+   * Configures the expected type of response the endpoint will return. Currently supported:
+   *<ul>
+   *   <li>
+   *     {@code json}
+   *     <p>
+   *       The response is appended to `_response` key of action's payload as parsed {@link JsonObject JsonObject}/{@link io.vertx.core.json.JsonArray JsonArray}.
+   *     </p>
+   *   </li>
+   *   <li>
+   *     {@code raw}
+   *     <p>
+   *       The response is appended to `_response` key of action's payload as a {@link JsonObject JsonObject} in the following format:
+   *       <pre>
+   *         {
+   *           "body": "<i>raw response body of the endpoint</i>",
+   *           "contentType": "<i>{@code Content-Type} header value (if any)</i>",
+   *           "encoding": "<i>encoding of the body (if could be determined)</i>
+   *         }
+   *       </pre>
+   *     </p>
+   *   </li>
+   *</ul>
+   *
+   * By default it is set to {@code json}.
+   *
+   * @param responseType - expected response type
+   * @return a reference to this, so the API can be used fluently
+   */
+  public HttpActionOptions setResponseType(String responseType) {
```

Comment:
Please check web client response predicates. Maybe we can configure them and:
- validate the content type
- save the response body in `_response` or in `_response.body`.

Reply:
Sounds cool, but the [default JSON predicate](https://vertx.io/docs/apidocs/io/vertx/ext/web/client/predicate/ResponsePredicate.html#JSON) is validating only content-type header. Expecting the external API to pass the correct headers might be quite a painful limitation in my opinion. We could implement our own one by try-parsing the response body but that would lead to exception-driven behaviour.